presto-product-tests/bin/run_on_docker.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash -ex
+
+# http://stackoverflow.com/questions/3572030/bash-script-absolute-path-with-osx
+function absolutepath() {
+    [[ $1 = /* ]] && echo "$1" || echo "$PWD/${1#./}"
+}
+
+ENVIRONMENT=$1
+
+if [[ "$ENVIRONMENT" != "singlenode" && "$ENVIRONMENT" != "distributed" ]]; then
+   echo "Usage: run_on_docker.sh <singlenode|distributed> <product test args>"
+   exit 1
+fi
+
+shift 1
+
+SCRIPT_DIR=$(dirname $(absolutepath "$0"))
+PRODUCT_TESTS_ROOT="${SCRIPT_DIR}/.."
+PROJECT_ROOT="${PRODUCT_TESTS_ROOT}/.."
+DOCKER_COMPOSE_LOCATION="${PRODUCT_TESTS_ROOT}/conf/docker/${ENVIRONMENT}/docker-compose.yml"
+DOCKER_LOGS_LOCATION="/tmp/presto_docker_logs"
```

Comment:
Are you talking about `absolutepath` function? It must be declared on top because it is called for determining `SCRIPT_DIR`. Do you think i should move all the other functions on top?